File: backend/microservice/src/level/objects/social/Favorite.scala
package microservice.level.objects.social

import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}

/** 玩家收藏关卡的记录。
  *
  * 字段说明：
  *   - id / levelId / userId：收藏标识与关联
  *   - createdAt：收藏时间
  */
final case class Favorite(
  id: String,
  levelId: String,
  userId: String,
  createdAt: String
)

object Favorite {
  implicit val encoder: Encoder[Favorite] = deriveEncoder
  implicit val decoder: Decoder[Favorite] = deriveDecoder
}

File: frontend/src/objects/level/social/favorite.ts
import { z } from "zod";

export const FavoriteSchema = z.object({
  id: z.string().min(1),
  levelId: z.string().min(1),
  userId: z.string().min(1),
  createdAt: z.string(),
});

export type Favorite = z.infer<typeof FavoriteSchema>;
